tests/shaping_impl.rs

```diff
@@ -0,0 +1,54 @@
+pub fn shape(font: &str, font_size: usize, variations: &[(&str, f32)], input: &str) -> String {
+    let file = std::fs::read(font).unwrap();
+    let font = swash::FontRef::from_offset(&file, 0).unwrap();
+    let mut context = swash::shape::ShapeContext::new();
+    let builder = context
+        .builder(font)
```

Comment:
We'll need to guess the script here so that the shaper picks up the correct tables.

```suggestion
    use swash::text::{Script, Codepoint as _};
    let file = std::fs::read(font).unwrap();
    let font = swash::FontRef::from_offset(&file, 0).unwrap();
    let script = input
        .chars()
        .map(|ch| ch.script())
        .find(|&script| {
            script != Script::Unknown && script != Script::Inherited && script != Script::Common
        })
        .unwrap_or(Script::Latin);    
    let mut context = swash::shape::ShapeContext::new();
    let builder = context
        .builder(font)
        .script(script)
```